### PR TITLE
fix for RN 0.40

### PR DESCRIPTION
**Pull request by jedt**
For more info: https://github.com/facebook/react-native/wiki/Breaking-Changes#namespace-all-header-imports-to-react-e1577d---javache

### CODE REVIEW

**Comment by natorojr** [comment]
Dear @yamill,

First of all, thank you for your contributions to the open source community.

I can understand that you may be preoccupied with your other projects, but I would greatly appreciate if you could merge in this PR, which includes the changes needed to support React Native 0.40+. As you can see, the changes are very straightforward (2 lines).

I'm using this library in a few key projects, but am unable to upgrade to React Native 0.40 due to these breaking changes. I'd prefer not to fork the project, as I am personally against fragmentation / duplication in the community and would rather continue advocating your library and the excellent work you've already done.

Thank you in advance for your consideration.

**Comment by manask88** [comment]
+1

**Comment by thomasobrien99** [comment]
+1

**Comment by realjamie** [comment]
+1

**Comment by carolinempapp** [comment]
+1

**Comment by danlopez** [comment]
👍

**Comment by georgiana-gligor** [comment]
If you need to use RN 0.40 asap, there's a quick solution that you can use in your project today, until the author merges this pull request.

Go to `package.json` and search for the dependency that looks similar to

```
"react-native-orientation": "^1.17.0",
```

Replace it with:
```
"react-native-orientation": "git+https://github.com/jedt/react-native-orientation.git",
```

This will effectively point your package manager to the pull request author's `master` branch.

**Comment by almirfilho** [comment]
+1

**Comment by manask88** [comment]
this is great @almirfilho . When can we expect this PR to be merged and a new npm package?  If you guys are overwhelmed with other projects, I can offer my self to help here :)

File: iOS/RCTOrientation/Orientation.h [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 #import <Foundation/Foundation.h>
 #import <UIKit/UIKit.h>
-#import "RCTBridgeModule.h"
+#import <React/RCTBridgeModule.h>
```
Comment (max-mykhailenko) [inline iOS/RCTOrientation/Orientation.h]: You should use conditional import like in https://github.com/inProgress-team/react-native-youtube/pull/104/commits/9cef77452cd72cc3c3b30b3895c39d5560c51088#diff-c25f51049843854bc862649b7c6e9f58R12

Comment (manask88) [inline iOS/RCTOrientation/Orientation.h]: @jedt , I think max suggestion is good. That way, the library can be backwards compatible. Do you think you can update your PR soon? If you need any help, let me know!

**Comment by ugiacoman** [comment]
Can confirm this PR works with RN v "0.42.3". Please merge!

**Comment by ugiacoman** [comment]
@fungilation Yes, I do think someone should make an official fork. Someone who is active and willing to merge PRs haha

**Comment by andrerfneves** [comment]
5a16ac8a3f96e12969d86a1dd1f6d548efc213b2 was merged and solves compatibility issues with new RN versions.

**Comment by fungilation** [comment]
@andrerfneves, should merge this patch instead with eliminating unnecessary HEADER_SEARCH_PATHS.

**Comment by andrerfneves** [comment]
@fungilation good catch. Just pushed a hotfix that solves this issue: 01feaaaeadc62d02d26adbdf4633c3c92c88429c

**Comment by fungilation** [comment]
Great. Even better if it gets bumped to npm 😉

**Comment by andrerfneves** [comment]
I don't currently have access to the NPM package to publish it myself. I'll work with the package owner to get him to publish it when the updates have all been completed. @fungilation for now, let's keep the package pointed to `master` on github. I'll make sure to add a note on README as well.

**Comment by fungilation** [comment]
Awesome. Thanks for bringing this package back to life!

**Pull request closed without merging**